test/test_cursor.py

```diff
@@ -1466,6 +1466,69 @@ def test_find_raw(self):
         self.assertEqual(1, len(batches))
         self.assertEqual(docs, decode_all(batches[0]))
 
+    @client_context.require_transactions
+    def test_find_raw_transaction(self):
+        c = self.db.test
+        c.drop()
+        docs = [{'_id': i, 'x': 3.0 * i} for i in range(10)]
+        c.insert_many(docs)
+
+        listener = EventListener()
+        client = rs_or_single_client(event_listeners=[listener])
+        with client.start_session() as session:
+            with session.start_transaction():
+                batches = list(client.db.test.find_raw_batches(
+                    session=session).sort('_id'))
+        self.assertEqual(1, len(batches))
+        self.assertEqual(docs, decode_all(batches[0]))
+
+        command = listener.results['started'][0].command
+        self.assertEqual(listener.results['started'][0].command_name, 'find')
+        self.assertEqual(command['$clusterTime'],
+                         decode_all(session.cluster_time.raw)[0])
```

Comment:
Weird, `session.cluster_time` is a RawBSONDocument?

Reply:
Yes... :/

Reply:
This is because of exhaust + OP_MSG changes.